Add etcd discovery flags (#1960)
sensu/sensu-docs: content/sensu-go/5.16/reference/backend.md

etcd-client-cert-auth: true{{< /highlight >}}

| etcd-discovery        |      |
------------------------|------
description             | Exposes [etcd's embedded auto-discovery features][17]. Attempts to use [etcd discovery][18] to get the cluster configuration.
type                    | String
default                 | ""
example                 | {{< highlight shell >}}# Command line example

> sensu-backend start --etcd-discovery https://10.0.0.1:2380

Author: ```suggestion
sensu-backend start --etcd-discovery https://discovery.etcd.io/3e86b59982e49066c5d813af1c2e2579cbf573de
```

# /etc/sensu/backend.yml example
etcd-discovery:

>   - https://10.0.0.1:2380

Reviewer: For this discovery method, I think you should reuse one of the example URLs used in the Etcd docs, e.g.: `https://discovery.etcd.io/3e86b59982e49066c5d813af1c2e2579cbf573de`.

>   - https://10.0.0.1:2380

Author: ```suggestion
  - https://discovery.etcd.io/3e86b59982e49066c5d813af1c2e2579cbf573de
```

{{< /highlight >}}

| etcd-discovery-srv    |      |
------------------------|------
description             | Exposes [etcd's embedded auto-discovery features][17]. Attempts to use a [DNS SRV][19] record to get the cluster configuration.
type                    | String
default                 | ""
example                 | {{< highlight shell >}}# Command line example

> sensu-backend start --etcd-discovery-srv https://10.0.0.1:2380

Reviewer: In the case of DNS SRV, the argument is simply a domain name, such as `example.org` or `test.com`. Etcd then uses the [SRV record](https://en.wikipedia.org/wiki/SRV_record) of that domain name to discover the cluster configuration.

> sensu-backend start --etcd-discovery-srv https://10.0.0.1:2380

Author: ```suggestion
sensu-backend start --etcd-discovery-srv example.org
```

# /etc/sensu/backend.yml example
etcd-discovery-srv:

>   - https://10.0.0.1:2380

Author: ```suggestion
  - example.org
```

| keepalived-buffer-size |      |
-----------------------|------

> description            | Number of incoming keepalives that can be buffered before beinghttps://etcd.io/docs/v3.3.12/op-guide/clustering/#dns-discovery processed by a keepalived worker. _WARNING: Modify with caution. Increasing this value may result in higher memory usage._

Reviewer: I'm assuming this hunk should be discarded?

> description            | Number of incoming keepalives that can be buffered before beinghttps://etcd.io/docs/v3.3.12/op-guide/clustering/#dns-discovery processed by a keepalived worker. _WARNING: Modify with caution. Increasing this value may result in higher memory usage._

Author: ```suggestion
description            | Number of incoming keepalives that can be buffered before being processed by a keepalived worker. _WARNING: Modify with caution. Increasing this value may result in higher memory usage._
```

Author: Yes, I must have pasted it there without realizing. Thank you!
